ast/TreeEquality.cc

```diff
@@ -302,39 +317,56 @@ bool structurallyEqual(const void *avoid, const Tag tag, const void *tree, const
     }
 }
 
-bool structurallyEqual(const void *avoid, const ExpressionPtr &tree, const ExpressionPtr &other, bool root) {
-    ENFORCE(tree != nullptr);
-    ENFORCE(tree != nullptr);
+bool structurallyEqual(const core::GlobalState &gs, const void *avoid, const ExpressionPtr &tree,
+                       const ExpressionPtr &other, const std::shared_ptr<spdlog::logger> logger,
+                       const core::Loc selectionLoc, bool root) {
```

Comment:
selectionLoc is a bit leaky—it implies that you'd only ever want to call this method from LSP, when this method is implemented as a non-LSP helper.

Do we actually need the selectionLoc? I was thinking that if we had the file path & contents, it would not be hard to drop into a debugger and find which parts of a file had a `nullptr` tree in them somewhere.

We could also probably reconstitute the selectionLoc from the other log lines that would appear before this log line, e.g. the log lines for every LSP request/response.